concrete/src/Cache/Adapter/ZendCacheDriver.php

```diff
@@ -27,27 +27,78 @@
 class ZendCacheDriver extends AbstractAdapter implements StorageInterface, FlushableInterface
 {
     /**
-     * @var string Name of the cache being used
+     * @var string Name of the cache being used when the localization is fully initialized
      */
-    private $cacheName;
+    private $finalCacheName;
+
+    /**
+     * @var \Concrete\Core\Application\Application
+     */
+    protected $app;
+
+    /**
+     * @var \Concrete\Core\Config\Repository\Repository
+     */
+    protected $config;
+
+    /**
+     * Is the localization system ready?
+     *
+     * @var bool
+     */
+    private $localizationReady;
 
     /**
      * @var int Number of seconds to consider the cache fresh before it expires
      */
     protected $cacheLifetime;
 
     /**
-     * @param string $cacheName Name of the cache being used. Defaults to cache.
-     * @param int $cacheLifetime Number of seconds to consider the cache fresh before it expires.
+     * @param string $finalCacheName Name of the cache being used when the localization is fully initialized. Defaults to cache
+     * @param int $cacheLifetime Number of seconds to consider the cache fresh before it expires
      */
-    public function __construct($cacheName = 'cache', $cacheLifetime = null)
+    public function __construct($finalCacheName = 'cache', $cacheLifetime = null)
     {
         parent::__construct();
-
-        $this->cacheName = $cacheName;
+        $this->setApplication(Application::getFacadeApplication());
+        $this->localizationReady = false;
+        $this->finalCacheName = $finalCacheName;
         $this->cacheLifetime = $cacheLifetime;
     }
 
+    public function setApplication(\Concrete\Core\Application\Application $app)
+    {
+        $this->app = $app;
+        $this->config = $app->make('config');
+    }
+
+    /**
+     * Is the localization system ready?
+     *
+     * @return bool
+     */
+    protected function isLocalizationReady()
+    {
+        if (!$this->localizationReady) {
+            if ($this->config->get('app.bootstrap.packages_loaded') === true) {
+                $this->localizationReady = true;
+            }
+        }
+
+        return $this->localizationReady;
+    }
+
+    /**
+     * @return \Concrete\Core\Cache\Cache|null
+     */
+    protected function getCache()
+    {
+        if ($this->isLocalizationReady()) {
+            return $this->app->make($this->finalCacheName);
+        } else {
+            return null;
```

Comment:
Maybe because of many years programming in other languages, as for myself I prefer writing `return null` instead of just `return`.  I know they lead to the same result, but they are conceptually different...
Think for instance at:
- a function like `echo` that never returns anything. 
- a function that parses a string and returns an int if it's valid or null otherwise, 

I'd only use `return` alone in the first case, when functions are not supposed to never return anything.
That's why I suggested [this](https://github.com/concrete5/concrete5/blob/dfc9c85284411d09a6fb24f7b9785aab995db11f/.php_cs#L15-L16) for our `.php_cs`